cmd/jujud/agent/machine/manifolds.go

```diff
@@ -151,6 +151,9 @@ type ManifoldsConfig struct {
 	// UpdateLoggerConfig is a function that will save the specified
 	// config value as the logging config in the agent.conf file.
 	UpdateLoggerConfig func(string) error
+
+	// Reporter provides upgradesteps.StatusSetter.
+	Reporter func(apiConn api.Connection) (upgradesteps.StatusSetter, error)
```

Comment:
In this context, I don't think "Reporter" is descriptive enough. We have a PubSubReporter, and we could end up with other reporters. I suggest calling this "NewAgentStatusSetter".

Reply:
renamed as suggested